cores/esp8266/core_esp8266_main.cpp

```diff
@@ -57,8 +56,8 @@ cont_t* g_pcont __attribute__((section(".noinit")));
 /* Event queue used by the main (arduino) task */
 static os_event_t s_loop_queue[LOOP_QUEUE_SIZE];
 
-/* Used to implement optimistic_yield */
-static uint32_t s_micros_at_task_start;
+/* Used to implement __optimistic_yield */
+static uint32_t s_cycles_since_yield_start;
```

Comment:
Suggest a better name for this would be `s_cycles_at_yield_start`.  It is holding an absolute cycle count(at), not a delta(since).